src/Avalonia.Visuals/Animation/CrossFade.cs

```diff
@@ -97,49 +99,39 @@ public Easing FadeOutEasing
             set => _fadeOutAnimation.Easing = value;
         }
 
-        /// <summary>
-        /// Starts the animation.
-        /// </summary>
-        /// <param name="from">
-        /// The control that is being transitioned away from. May be null.
-        /// </param>
-        /// <param name="to">
-        /// The control that is being transitioned to. May be null.
-        /// </param>
-        /// <returns>
-        /// A <see cref="Task"/> that tracks the progress of the animation.
-        /// </returns>
-        public async Task Start(Visual from, Visual to)
+        /// <inheritdoc cref="Start(Visual, Visual, CancellationToken)" />
+        public async Task Start(Visual from, Visual to, CancellationToken cancellationToken)
         {
-            var tasks = new List<Task>();
-
-            if (to != null)
-            {
-                to.Opacity = 0;
-            }
-
-            if (from != null)
+            if (cancellationToken.IsCancellationRequested)
             {
-                tasks.Add(_fadeOutAnimation.RunAsync(from));
+                return;
             }
 
-            if (to != null)
+            var tasks = new List<Task>();
```

Comment:
this part looks strange to me. When we have only 1 task to await no need to use Task.WhenAll. When we have 2 tasks don't create a list, Task.WhenAll have overload which accepts params Task[].

Reply:
I wasn't changing previous code here, though it makes sense to optimize it for a bit like you said. Still it will also make code a bit more complex as well.

Reply:
sry haven't realized this code previously existed